### PR TITLE
[risk=low][RW-6485] Fix a bug in UpdateCdrConfig and add tests

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/tools/cdrconfig/CdrConfigVOMapper.java [hunk 1]
```diff
@@ -26,14 +29,24 @@
     return toDbTiers(cdrConfig.accessTiers);
   }
 
-  // currently unambiguous, but adding the qualifiedByName to future-proof
-  @Mapping(source = "accessTier", target = "accessTier", qualifiedByName = "findOneByShortName")
+  // MapStruct gets the accessTier mapping wrong, by choosing the wrong AccessTierDao method
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/tools/cdrconfig/CdrConfigVOMapper.java:32]: MapStruct was choosing "findAll()" for some reason, which resulted in retrieving the existing list of access tiers and THEN adding new ones from the file to the list

Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/tools/cdrconfig/CdrConfigVOMapper.java:32]: qualifiedByName did not work in the way I assumed

File: api/tools/src/main/java/org/pmiops/workbench/tools/cdrconfig/UpdateCdrConfig.java [hunk 2]
```diff
@@ -241,7 +241,7 @@ private void updateDB(
       currentCdrVersions.put(cdrVersion.getCdrVersionId(), cdrVersion);
     }
 
-    List<DbCdrVersion> cdrVersions = cdrConfigMapper.cdrVersions(cdrConfig);
+    List<DbCdrVersion> cdrVersions = cdrConfigMapper.cdrVersions(cdrConfig, accessTierDao);
```
Comment (calbach) [inline api/tools/src/main/java/org/pmiops/workbench/tools/cdrconfig/UpdateCdrConfig.java:244]: To clarify - the issue I saw is that there were duplicate elements in the `accessTiers` array, i.e. L194. Is that consistent with the fix here? I can't really work backwards to this, but it seems the cause was some odd interaction between the mapper and dao, so it's possible this fixes it.

To verify, when you run dev-up, logs should indicate that `accessTiers` are unchanged (unless it's your first time running dev-up).

**Review by calbach — APPROVED** [review]
Thanks for the tests

File: api/tools/src/main/java/org/pmiops/workbench/tools/cdrconfig/UpdateCdrConfig.java (hunk 2 above)
Comment (jmthibault79) [reply api/tools/src/main/java/org/pmiops/workbench/tools/cdrconfig/UpdateCdrConfig.java:244]: yes - I do see accessTiers are unchanged on dev-up

Comment (jmthibault79) [reply api/tools/src/main/java/org/pmiops/workbench/tools/cdrconfig/UpdateCdrConfig.java:244]: The accessTiers array is populated via `mapper.toDbTiers()` which is where the problem occurred.  So this does make sense.

**Pull request merged**